### ISSUE TITLE
Single product gallery repetitive for screen readers.

### ISSUE THREAD

**Issue by melek** [woothemes/woocommerce#31955]
### Prerequisites

- [X] I have carried out troubleshooting steps and I believe I have found a bug.
- [X] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

Right now, the single product image gallery is spoken before the product title, price, and product data on the Single Product Page when using a screen reader:

* It is inconvenient for blind users to have to flip through descriptions of photographs before getting to the product title and price.
* Each image is read aloud to even if there is no 'Alt' text defined for the image in the media library.
* Each image is read aloud a second time in the thumbnail row, this time as 'Image 1/3', etc.; so is the 'zoom widget' (magnifying glass).

This was brought to our attention by a customer with a blind client mentioned that including multiple product images makes the price and product description annoying to reach (4800910-zen).

### Expected behavior

I believe to be inclusive of screen reader users:

* Images with no Alt text should not be read aloud.
* Elements only relevant to visual users like the thumbnail gallery and zoom widget should not be read aloud.

Changing the order of element so the photo gallery is not above the title and price in the document flow would require a pretty major change to the templates, so I'd say outside the scope of this report. It may be worth thinking about in a larger effort like Accessibility Issues (#24185).

### Actual behavior

As per the bug description; single product images without an alt tag are rendered by screen readers, as well as the image thumbnails and zoom widget.

Here is a simulation with an extension that visualizes the screen reader content:

![Screenshot](https://uploads.linear.app/f629fcde-6f9b-48e8-8af3-60eed8823bff/7d63264d-0cc9-44c2-accb-f94a7fdf54cc/7d2ae34a-ddca-4d5d-b287-9ad9002e9eb9?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiL2Y2MjlmY2RlLTZmOWItNDhlOC04YWYzLTYwZWVkODgyM2JmZi83ZDYzMjY0ZC0wY2M5LTQ0YzItYWNjYi1mOTRhN2ZkZjU0Y2MvN2QyYWUzNGEtZGRjYS00ZDVkLWIyODctOWFkOTAwMmU5ZWI5IiwiaWF0IjoxNzcyNDQyNDM5LCJleHAiOjE4MDQwMTI5OTl9.R6M0Z8GLXNr1pvmo4tJT68xu0ZXaoqKIOIneSxoLtBM)

*If you do not see the image, please *[*click here*](<https://d.pr/i/KitQ04>).

Using Voiceover on Mac, I had to skip through each image two times - **even when no Alt attribute was set**:

![Screenshot](https://uploads.linear.app/f629fcde-6f9b-48e8-8af3-60eed8823bff/bffcfbee-87ee-43ac-bfab-32da3fe8b878/6661c476-6dbb-466b-ba40-6125deb9ffeb?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiL2Y2MjlmY2RlLTZmOWItNDhlOC04YWYzLTYwZWVkODgyM2JmZi9iZmZjZmJlZS04N2VlLTQzYWMtYmZhYi0zMmRhM2ZlOGI4NzgvNjY2MWM0NzYtNmRiYi00NjZiLWJhNDAtNjEyNWRlYjlmZmViIiwiaWF0IjoxNzcyNDQyNDM5LCJleHAiOjE4MDQwMTI5OTl9.NuFJb-OqeB_6b_ehRegnTP4zwWcU4XhAiz0AvPvDZAc)

*If you do not see the image, please *[*click here*](<https://d.pr/i/upAqWn>).

### Steps to reproduce

1. View any single product page with multiple product images. Have no alt attributes set on these images.
2. Engage screen reader software.
3. Navigate through the page, noting how the photo gallery is rendered - each image is spoken aloud twice.

### WordPress Environment

WC Version: 6.2.0<br>WP Version: 5.9<br>Theme Name: Storefront<br>Version: 3.9.1

### Isolating the problem

- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](<https://woocommerce.com/storefront/>).
- [X] I can reproduce this bug consistently using the steps above.

**Comment by woocommercebot** [comment]
We are adding the `status: needs reproduction` label to this issue to try reproduce it on the   current released version of WooCommerce.

  Thank you for your patience.

**Comment by swatipawarGS** [comment]
Hi @melek,

Thank you for taking the time to report this bug, we really appreciate your help. We are able to reproduce the issue on our end using WooCommerce 6.2.0 version and WordPress 5.9 version. 

Please find below screenshot for reference :

https://user-images.githubusercontent.com/94531721/155352966-6cc4879c-d4c6-48aa-8346-501a05213420.mp4

Priority of this issue will be assigned shortly after we’d go through a round of prioritization. Please note it may take a couple of days.

**Comment by amesplant** [comment]
I am going to add this to my list of updates. Looks like several things need to be addressed and I want to outline them here for clear communication.

1. The order of the DOM should first read the Product Title before any other detail
2. Fix the way that the images are announced
     a. when the page is read sequentially, the images are problematic in the way that they change as if they are being navigated through when the screen reader is reading them
     b. fix the tab order on the images, when using the tab key to navigate the links and landing on the image, the next tab simply swaps out the image vs. tabbing to the thumbnails; while this might not be an issues for non-visual users, it is an issue for users who are visual and also use a screen reader.
3. Fix the missing Focus on the image links
4. The product images are appearing 2x within the Rotor for VoiceOver given the way that PhotoSwipe is managing the slide show aspect.

**Comment by Manussakis** [comment]
FlexSlider causes most of the accessibility issues reported here. However, the plugin is [no longer under active development](https://github.com/woocommerce/FlexSlider?tab=readme-ov-file#no-longer-under-active-development) and there is an [ongoing discussion](https://github.com/woocommerce/woocommerce/discussions/48302) to determine the updates the new product slider should get, such as jQuery removal and accessibility improvements.

According to my investigation, only two issues flagged here are not caused by FlexSlider.
### Issue 1 
> It is inconvenient for no-visual users to have to flip through descriptions of photographs before getting to the product title and price.

I believe we don't need to address it. The reasons are:
    - This is especially a problem if the product images are announced twice, but updating the slider should solve it.
    - The product name is present in the page title that should be announced as the page loads.
    - Users can leverage the screen reader functionality to jump directly to the main heading of the page, which is the Product name. From there, it should be easy to get other product information such as the price.
    - This could be fixed by switching the DOM order of the product gallery and the product information containers. Then, re-position them with `row-reverse`/`column-reverse` flex directions. The tradeoff is that users would reach the "Add to cart" button before becoming aware of the product gallery's existence.

### Issue 2
> Elements only relevant to visual users like the [...] zoom widget should not be read aloud.
- The zoomed image is no longer announced once the [[Accessibility] Hide zoomed product images for screen readers.](https://github.com/woocommerce/woocommerce/pull/50003) PR was merged. So this is already fixed.

@amesplant did a great job putting together [the fixes the slide needs](https://github.com/woocommerce/woocommerce/issues/31955#issuecomment-2251163210) (only the first item of their list is not related to the slider issue).

Fixing these issues without touching on the FlexSlider means putting a huge effort into working around the plugin's accessibility issue. In addition to that, once the product slider is updated, most of the effort with these workarounds would have to be reverted.

I suggest linking this issue to the FlexSlider removal/update issue once it's created instead of addressing it separately.

cc: @woocommerce/woo-fse

**Comment by vikrampm1** [comment]
Moving this to the backlog until further notice.

**Comment by vikrampm1** [comment]
We have removed this from our backlog and unassigned ourselves, so that someone from Woo can take it forward. As noted above, this issue needs to be tackled in the FlexSlider extension.
